fenix/app/src/main/java/org/mozilla/fenix/debugsettings/store/DebugDrawerAction.kt

```diff
@@ -0,0 +1,23 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.debugsettings.store
+
+import mozilla.components.lib.state.Action
+
+/**
+ * [Action] implementation related to [DebugDrawerStore].
+ */
+sealed class DebugDrawerAction : Action {
+
+    /**
+     * [DebugDrawerAction] fired when the user opens the drawer.
+     */
+    object DrawerOpened : DebugDrawerAction()
+
+    /**
+     * [DebugDrawerAction] fired when the user closes the drawer.
+     */
+    object DrawerClosed : DebugDrawerAction()
+}
```

Comment:
I won't ask you to change it again, but I do think this is an interesting case study in the verbosity/clarity spectrum. IMO this is tending a little more towards boilerplate-y

Reply:
I'm by no means committed to this. In case you didn't catch the convo before I collapsed it, this change is a result of this [comment thread](https://github.com/mozilla-mobile/firefox-android/pull/4842#discussion_r1430430963). Since this is still a draft technically, I felt it was at least worth exploring (wording withstanding).

Reply:
@t-p-white what's your take having seen the before/after now? 

Personally, I'm leaning towards going back to one action + a boolean to keep things simple and concise

Reply:
@MozillaNoah Thanks for trying this out 👍 

Personally, I prefer this method of having the separate actions and using enums, I'm not sold on a Boolean as an appropriate type for open/close - as you found an enum is also how the `Material` `DrawerState` defines the values, so aligning with that design is no bad thing. IMO this just removes any ambiguity with each object having a single responsibility and should make it more open for extension, for example if you want to add a parameter in the future for just 'open' this method makes that an easy change

Reply:
> enum is also how the Material DrawerState defines the values
Actually, couldn't we just use the existing `DrawerValue` enum?

Reply:
We could. The only trade-off is that the `DebugDrawerState` would now be coupled to that Material class/library and it isn't extensible, but maybe that's overengineering it when we only know what we need right now anyway.

Reply:
Yeah that's a good point, certainly doesn't hurt to roll our own 👍

Reply:
I think we want to keep platform and external library types out of our state unless absolutely necessary to avoid coupling.  

I'm also not set on using a `Boolean` necessarily, as I do think there is value in well-defined types. I could've clarified further with my last comment, but I do think a single `Toggled` action that includes the updated state simplifies callsites and the reducer. 

Further, the nested types become fairly wordy, but that could be changed with a manual import if you wanted. For example:
```
DebugDrawerState.DrawerValue.Closed

Reply:
I'll keep what we have for now. Maybe this can be a 5 minute topic in show and tell?

![image](https://github.com/mozilla-mobile/firefox-android/assets/87384386/9493748b-9555-4e78-8bc7-bfc19b81f695)

![image](https://github.com/mozilla-mobile/firefox-android/assets/87384386/91435254-9062-473a-80f2-e832510ed856)


This is about as clean, single responsibility as it gets

Reply:
Sounds good @MozillaNoah! So I guess this comes down to:
1. Separate Open/Close actions as you have it here
2. A toggle drawer action that takes the Open/Close enum type
?

My personal preference is having the explicit Open/Close actions:
1.  it's super clear to any users what the action is doing
2. I think this lends itself better for extension

But I can see we already have 'toggle' type actions in the code.